Replicator/Pusher.cc

```diff
@@ -512,26 +512,76 @@ namespace litecore::repl {
         Worker::_connectionClosed();
     }
 
-    bool Pusher::isBusy() const {
-        return Worker::computeActivityLevel() == kC4Busy || (_started && (!_caughtUp || !_continuousCaughtUp))
-               || _changeListsInFlight > 0 || _revisionsInFlight > 0 || _blobsInFlight > 0 || !_revQueue.empty()
-               || !_pushingDocs.empty() || _revisionBytesAwaitingReply > 0;
+    bool Pusher::isBusy(std::string* reason) const {
+        std::string reason0;
```

Comment:
Better than `k` but still not a great variable name.

Reply:
It's the reason as we delegate the decision to other functions. It may not be used depending on whether it is overwritten later in the decision path. Maybe potentialReason?

Reply:
Given that it comes from the Worker parent class something like `innerReason` or `parentReason` seems more clear.